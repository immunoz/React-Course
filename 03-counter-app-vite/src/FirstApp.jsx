import PropTypes from 'prop-types';
import { getHeroById } from '../../02-intro-javascript/src/bases/08-imp-exp';

const FirstApp = ( { title, subtitle, name } ) => {

  return (
    <>
        <h1 data-testid='test-title'>{ title }</h1>
        <span>{ subtitle }</span>
        <span>{ subtitle }</span>
        <br/>
        { /* <code> { JSON.stringify(getHeroById(1)) } </code> */ }
        <p>{ name }</p>
    </>
  )
}

export default FirstApp

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string,
}

FirstApp.defaultProps = {
  title: 'No hay título',
  subtitle: 'No hay subtítulo',
}

/*
Notes:
- Default props always go before the proptypes
- Be careful with the default values and the proptypes. If a property with a non accepted value is received on the default values, an error will be thrown.
 */