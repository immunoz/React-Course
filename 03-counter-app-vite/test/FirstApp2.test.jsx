import { render, screen } from "@testing-library/react"
import FirstApp from "../src/FirstApp";

describe('Testing <FirstApp />', () => {

    const title = 'Hello World';
    const subtitle = 'Hello World subtitle';
    
    test('should match snapshot', () => {

        const { container } = render( <FirstApp title={title} /> );
        expect( container ).toMatchSnapshot();
      
    });

    test('should show the message "Hello World"', () => {
      
        render( <FirstApp title={title} /> );
        expect( screen.getByText(title) ).toBeTruthy();

    });

    test('should show the title in a h1', () => {
      
        render( <FirstApp title={title} /> );
        expect( screen.getByRole('heading', { level: 1}).innerHTML ).toContain(title);
    });

    test('should show subtitle sent by props', () => {
      
        render(
            <FirstApp
                title={ title }
                subtitle={ subtitle}
            />
        );

        expect( screen.getAllByText(subtitle).length ).toBe(2);
        
    });
    
});
