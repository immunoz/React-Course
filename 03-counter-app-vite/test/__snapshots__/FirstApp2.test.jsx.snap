// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <FirstApp /> should match snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hello World
  </h1>
  <span>
    No hay subtítulo
  </span>
  <span>
    No hay subtítulo
  </span>
  <br />
  <p />
</div>
`;
